src/seedu/addressbook/data/person/Address.java

```diff
@@ -37,22 +58,69 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
+        return blockData.getBlock() + " " + streetData.getStreet() + " " + unitData.getUnit() + " " + postalCodeData.getPostalCode();
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                && this.toString().equals(((Address) other).toString())); // state check
     }
 
     @Override
     public int hashCode() {
-        return value.hashCode();
+        return (blockData.getBlock() + " " + streetData.getStreet() + " " 
+                + unitData.getUnit() + " " + postalCodeData.getPostalCode()).hashCode();
     }
 
     public boolean isPrivate() {
         return isPrivate;
     }
+}
+
+/**
+ * Subclasses of the address class.
+ * @author sw
```

Comment:
You can omit the `@author` tag as we expect multiple developers to edit the code.